src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -781,23 +788,56 @@ public NamedOnnxValue GetNamedOnnxValue()
 
             private class NamedOnnxValueGetterVec<T> : INamedOnnxValueGetter
             {
+                private delegate NamedOnnxValue GetNamedOnnxVal();
+
                 private readonly ValueGetter<VBuffer<T>> _srcGetter;
                 private readonly OnnxShape _tensorShape;
                 private readonly string _colName;
                 private VBuffer<T> _vBuffer;
                 private VBuffer<T> _vBufferDense;
+                private readonly bool _isKnownSize;
+                private readonly int _denominator;
+                private readonly int _zeroIndex;
+                private readonly GetNamedOnnxVal _namedOnnxValueDelegate;
+
                 public NamedOnnxValueGetterVec(DataViewRow input, int colIndex, OnnxShape tensorShape)
                 {
                     _srcGetter = input.GetGetter<VBuffer<T>>(input.Schema[colIndex]);
+                    _isKnownSize = (input.Schema[colIndex].Type as VectorDataViewType).IsKnownSize;
                     _tensorShape = tensorShape;
                     _colName = input.Schema[colIndex].Name;
                     _vBuffer = default;
                     _vBufferDense = default;
+                    _denominator = _tensorShape.Where(x => x > 0).Aggregate((a, x) => a * x);
+                    _zeroIndex = _tensorShape.IndexOf(0);
+
+                    if (_isKnownSize)
```

Comment:
>_isKnownSize [](http://example.com/codeflow?start=24&length=12)

I am not seeing this used outside the constructor. why can't be a local variable instead of class field?

Reply:
Good catch. It can. Originally it was used elsewhere but I refactored and now its now. I'll pull it.